src/NuGet.Core/NuGet.Packaging/RuntimeModel/RuntimeGraph.cs

```diff
@@ -8,31 +8,33 @@
 using System.Linq;
 using NuGet.Shared;
 
+#nullable enable
+
 namespace NuGet.RuntimeModel
 {
-    public class RuntimeGraph : IEquatable<RuntimeGraph>
+    public sealed class RuntimeGraph : IEquatable<RuntimeGraph>
     {
-        private readonly ConcurrentDictionary<RuntimeCompatKey, bool> _areCompatible
-            = new ConcurrentDictionary<RuntimeCompatKey, bool>();
-
-        private readonly ConcurrentDictionary<string, HashSet<string>> _expandCache
-            = new ConcurrentDictionary<string, HashSet<string>>(StringComparer.Ordinal);
-
-        private readonly ConcurrentDictionary<RuntimeDependencyKey, List<RuntimePackageDependency>> _dependencyCache
-            = new ConcurrentDictionary<RuntimeDependencyKey, List<RuntimePackageDependency>>();
+        // These fields are null when IsEmpty is true
+        private readonly ConcurrentDictionary<RuntimeCompatKey, bool>? _areCompatible;
+        private readonly ConcurrentDictionary<string, HashSet<string>>? _expandCache;
+        private readonly ConcurrentDictionary<RuntimeDependencyKey, List<RuntimePackageDependency>>? _dependencyCache;
 
-        private HashSet<string> _packagesWithDependencies;
+        private HashSet<string>? _packagesWithDependencies;
 
         public static readonly string RuntimeGraphFileName = "runtime.json";
 
-        public static readonly RuntimeGraph Empty = new RuntimeGraph();
+        /// <summary>
+        /// Gets a singleton, immutable, empty instance of <see cref="RuntimeGraph"/>.
+        /// </summary>
+        public static readonly RuntimeGraph Empty = new();
```

Comment:
nit: personally I like not modifying lines for minor stylistic changes. It just makes the diff unnecessarily larger.